modules/web-console/frontend/app/modules/agent/AgentManager.service.ts

```diff
@@ -432,19 +424,11 @@ export default class AgentManager {
                     latch.resolve(evt.payload);
             });
 
-        this.wsConnection.pipe(
-            filter((isConnected) => isConnected),
-            first(),
-            // Await web socket connection to send request.
-            timeout(5000),
-            tap(() => this._sendWebSocketEvent(requestId, eventType, data)),
-            catchError(() => {
-                latch.reject({message: 'Connection to web server was lost'});
-
-                return EMPTY;
-            })
-        )
-        .subscribe(() => {});
+        try {
+            this._sendWebSocketEvent(requestId, eventType, data);
+        } catch (ignored) {
+            latch.reject({message: 'Connection to web server was not established yet'});
```

Comment:
Did you consider `_sendWebSocketEvent` might throw for other reasons too? The error has code you can check against.

Reply:
Doesn't matter which reason of exception we should return promise. Improved error message.